learn_qso_model.m

```diff
@@ -185,17 +186,17 @@
 % find empirical mean vector and center data
 mu = nanmean(rest_fluxes_div_exp1pz);
 centered_rest_fluxes = bsxfun(@minus, rest_fluxes_div_exp1pz, mu);
-clear('rest_fluxes', 'rest_fluxes_div_exp1pz');
+clear('rest_fluxes');
 
 % get top-k PCA vectors to initialize M
 [coefficients, ~, latent] = ...
     pca(centered_rest_fluxes, ...
         'numcomponents', k, ...
-        'rows',          'pairwise');
+        'rows',          'complete');
```

Comment:
Could you revert this?

Reply:
Yes I can revert it.

I think both pairwise and complete are fine. 

The problem is in the `spectrum_loss.m`

Reply:
Yup, I agree! complete sometimes fails if you send it particularly bitty data.

Reply:
I am using complete now and it runs fine

I remember complete assume there no missing data in the training set, which actually is not true for our data which have many NaNs.

Reply:
That's right - so pairwise is more appropriate, I think, but pairwise sometimes fails to return a positive definite result. As long as pairwise succeeds I think we prefer it (although it isn't really important)